httpcore5/src/test/java/org/apache/hc/core5/net/TestURIBuilder.java

```diff
@@ -444,14 +542,14 @@ public void testPathNoLeadingSlash() throws Exception {
                 .setScheme("ftp")
                 .setPath("blah")
                 .build();
-        Assert.assertThat(uri, CoreMatchers.equalTo(URI.create("ftp:/blah")));
+        MatcherAssert.assertThat(uri, CoreMatchers.equalTo(URI.create("ftp:/blah")));
```

Comment:
Wait a minute, why is there a slash missing? This looks wrong according to the RFC.

Reply:
@michael-o I am not sure what extra slash you are referring to. Authority separator? But there is no authority.

Reply:
If there is no authority, it should be `ftp:///blah`. You are missing proper `://` after the scheme.

Reply:
@michael-o I cannot agree here. Please take a look at the section 3
```
3. Syntax Components
The generic URI syntax consists of a hierarchical sequence of
components referred to as the scheme, authority, path, query, and
fragment.
URI = scheme ":" hier-part [ "?" query ] [ "#" fragment ]
hier-part = "//" authority path-abempty / path-absolute / path-rootless
```
URIs can have a scheme and a path components and nothing else. That way `ftp:/blah` is perfectly fine as well as `ftp:blah`. So it `ftp:///blah`

Reply:
I cannot see that `//` is optional anyhow. It is preceded by schema and colon. I Am not referring to authority and path-*.
Why do you think that `//` is optional?

Reply:
@michael-o The way I see it it is not required in case of `path-absolute`, `path-rootless` and `path-empty`.
`hier-part = "//" authority path-abempty / path-absolute / path-rootless / path-empty`

Reply:
My bad, you are right. I read section 3, formatting is key:

```
      URI         = scheme ":" hier-part [ "?" query ] [ "#" fragment ]

      hier-part   = "//" authority path-abempty
                  / path-absolute
                  / path-rootless
                  / path-empty
```

Syntactically, your test case is valid, but semantically not: https://tools.ietf.org/html/draft-yevstifeyev-ftp-uri-scheme-08#section-3.1

I consider using the `ftp` scheme as a bad example.

Reply:
@michael-o Agreed. I will tweak the test cases.